torch/csrc/jit/script/init.cpp

```diff
@@ -373,6 +357,35 @@ Resolver pythonResolver(ResolutionCallback rcb) {
   };
 }
 
+FunctionSchema getSchemaWithDefaults(
+    FunctionDefaults& default_args,
+    FunctionSchema schema) {
+  std::vector<Argument> new_args;
+  for (auto& arg : schema.arguments) {
+    auto it = default_args.find(arg.name);
+    if (it != default_args.end()) {
+      try {
+        IValue value = toIValue(it->second, arg.type);
+        new_args.push_back(
+            Argument(arg.name, arg.type, arg.N, value, arg.kwarg_only));
+      } catch (py::cast_error& e) {
+        throw ErrorReport()
```

Comment:
The error report needs a SourceRange, otherwise the error message will not be associated with a location in the code. You can get the SourceRange from the corresponding argument in the Def object.